format/nes/nes.go

```diff
@@ -0,0 +1,339 @@
+package nes
+
+import (
+	"embed"
+	"fmt"
+	"math"
+
+	"golang.org/x/text/encoding"
+
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/interp"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+//go:embed nes.jq
+//go:embed nes.md
+var nesFS embed.FS
+
+func init() {
+	interp.RegisterFormat(
+		format.NES,
+		&decode.Format{
+			Description: "NES 2.0",
+			Groups:      []*decode.Group{format.Probe},
+			DecodeFn:    decodeNES,
+		})
+	interp.RegisterFS(nesFS)
+}
+
+type nesContext struct {
+	nes20       bool
+	trainerSize uint64
+	prgROMSize  uint64
+	chrROMSize  uint64
+}
+
+func bitDesc(desc0 string, desc1 string) scalar.UintMap {
+	return scalar.UintMap{0: {Description: desc0}, 1: {Description: desc1}}
+}
+func boolDesc(desc0 string, desc1 string) scalar.BoolMap {
+	return scalar.BoolMap{false: {Description: desc0}, true: {Description: desc1}}
+}
+
+func getShiftSizeMapper() scalar.UintFn {
+	return scalar.UintFn(func(s scalar.Uint) (scalar.Uint, error) {
+		if s.Actual > 0 {
+			s.Sym = 64 << s.Actual
+		}
+		return s, nil
+	})
+}
```

Comment:
Yes, that looks much nicer. I had a function there before and it took the "64" in as an argument. But then the linter told me I was being dumb since I called it with "64" every time. =)